### PR TITLE
Launch/restructuring

### PR DESCRIPTION
**Pull request by bcsherma**
## Description
What does the pull request do? If it fixes a bug or resolves a feature request, be sure to link to that issue.

This PR restructures the launch docs to align with what was laid out in the discussion thread [for this PR](https://github.com/wandb/docodile/pull/151#issuecomment-1500466361).

Some content has been shuffled around.

- The quickstart has been changed to a much simpler example script and does not cover queue creation, instead we refer to the Starter queue path.
- The details about docker queue creation and additional details about launch on docker are now in the "Launch on Docker" page
- We now have a separate docs page dedicated to how to create queues.
- Some of the gifs were re-recorded because their content changed.
- Made some minor improvements to the agent documentation.
- Added a selector panel for describing the different job creation methods 

## Checklist
Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply. 

- [x] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [x] Build (`yarn build`) was run locally and successfully without errors or warnings.
- [x] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

File: docs/guides/launch/create-job.md
Comment (KyleGoyette) [inline docs/guides/launch/create-job.md:23]: Can we make the git one first, since it is the highest priority?

File: docs/guides/launch/create-job.md
Comment (KyleGoyette) [inline docs/guides/launch/create-job.md]: I don't think it's limited to GitHub, GitLab and BitBucket both also work. The true requirement is a remote git url

File: docs/guides/launch/create-job.md
Comment (KyleGoyette) [inline docs/guides/launch/create-job.md]: nit: The wording of when "wandb.init is called" could make users believe the job will exist on init, when it instead is created on close.

File: docs/guides/launch/create-job.md
Comment (KyleGoyette) [inline docs/guides/launch/create-job.md:51]: I think users may just need to see a snippet of `docker run -e WANDB_DOCKER=<image-name> image-name>` or something like that

File: docs/guides/launch/create-job.md
Comment (KyleGoyette) [inline docs/guides/launch/create-job.md]: Is this section necessary? Could we remove it? I'd rather users not need docs pointing this out

File: docs/guides/launch/create-job.md
Comment (KyleGoyette) [inline docs/guides/launch/create-job.md]: Could we just put this in like a small info blob or something?

File: docs/guides/launch/create-job.md
Comment (KyleGoyette) [inline docs/guides/launch/create-job.md]: "from the list"?

File: docs/guides/launch/create-job.md
Comment (KyleGoyette) [inline docs/guides/launch/create-job.md]: Some of this info seems repeated from the paragraph immediately above these tabs

File: docs/guides/launch/create-job.md
Comment (KyleGoyette) [inline docs/guides/launch/create-job.md]: Not limited to init, just any value in the config.

File: docs/guides/launch/create-job.md
Comment (KyleGoyette) [inline docs/guides/launch/create-job.md]: IMO this is unnecessary

File: docs/guides/launch/create-queue.md
Comment (KyleGoyette) [inline docs/guides/launch/create-queue.md]: "re" -> "are"

Comment (KyleGoyette) [reply docs/guides/launch/create-queue.md]: Maybe instead of saying "particular compute resource" something like "a compute resource associated with the queue"?

File: docs/guides/launch/create-queue.md
Comment (KyleGoyette) [inline docs/guides/launch/create-queue.md]: We say "compute resource" above, and "execution platform" here. How come?

File: docs/guides/launch/create-queue.md
Comment (KyleGoyette) [inline docs/guides/launch/create-queue.md]: demo or tutorial?

Comment (KyleGoyette) [reply docs/guides/launch/create-queue.md]: guide?

File: docs/guides/launch/create-queue.md
Comment (KyleGoyette) [inline docs/guides/launch/create-queue.md]: I wonder if we should make "used to specify arguments when run" should emphasize "resource arguments"

File: docs/guides/launch/docker.md
Comment (KyleGoyette) [inline docs/guides/launch/docker.md:1]: Will we have fully separate pages for EKS or GKE, Sagemaker, vertex etc?

File: docs/guides/launch/docker.md
Comment (KyleGoyette) [inline docs/guides/launch/docker.md]: "will run your launch agent.See" needs a space

Comment (KyleGoyette) [reply docs/guides/launch/docker.md]: Should we move the NVIDIA toolkit info down into the tip below?

File: docs/guides/launch/intro.md
Comment (KyleGoyette) [inline docs/guides/launch/intro.md:45]: Quick start instead of getting started?

File: docs/guides/launch/run-agent.md
Comment (KyleGoyette) [inline docs/guides/launch/run-agent.md:62]: In noop case the env and registry can both be missing

Comment (KyleGoyette) [reply docs/guides/launch/run-agent.md:62]: Is there a way we can specify the optionality of these in the yaml as well?

File: docs/guides/launch/run-agent.md
Comment (KyleGoyette) [inline docs/guides/launch/run-agent.md:143]: Below in the builders section can we indicate how to set your yaml up for a no build system explicitly, I think this will be somewhat common

File: docs/guides/launch/sweeps-on-launch.md
Comment (KyleGoyette) [inline docs/guides/launch/sweeps-on-launch.md]: using sweeps on launch?

**Review by KyleGoyette — COMMENTED** [review]
This is looking good. Do we intend on having a page for each environment? We have "Docker" will we have "Kubernetes"? Sagemaker? We may want to think about consolidating this info into Create a Queue (for the runner) and "Start an Agent" for the environment to start an agent

File: docs/guides/launch/create-job.md
Comment (ngrayluna) [inline docs/guides/launch/create-job.md]: Change this to "warning", we got a LOT of pushback for using the warning banner.

File: docs/guides/launch/create-job.md
Comment (ngrayluna) [inline docs/guides/launch/create-job.md:19]: Wondering if we can bring back an abridge version of the examples we had? (Minus the code in the script). https://github.com/wandb/docodile/commit/ef54209d1f99c44fb1b68f50b9e984962cee7aca#diff-310198a8a0fc02944502131b5d7a18bb74be5241ada41ddadc6fb1964dbb7935R84

File: docs/guides/launch/create-job.md
Comment (ngrayluna) [inline docs/guides/launch/create-job.md:72]: Nice. Thanks for making this section.

File: docs/guides/launch/docker.md
Comment (ngrayluna) [reply docs/guides/launch/docker.md:1]: Agreed. From there we can nest the Docker, SageMaker, and GKE into one subchapter.

File: docs/guides/launch/docker.md
Comment (ngrayluna) [inline docs/guides/launch/docker.md]: broken link

File: docs/guides/launch/getting-started.md
Comment (ngrayluna) [inline docs/guides/launch/getting-started.md]: This is funny, so I didn't call it a "quickstart" because we've had feedback from uppers on quickstarts not being "quick". So, for that reason, I called it a "Getting started" guide.

File: docs/guides/launch/getting-started.md
Comment (ngrayluna) [inline docs/guides/launch/getting-started.md]: Rename to "Queue your job"?

File: docs/guides/launch/launch-jobs.md
Comment (ngrayluna) [inline docs/guides/launch/launch-jobs.md]: "...to queue jobs...."

File: docs/guides/launch/create-job.md
Comment (bcsherma) [reply docs/guides/launch/create-job.md]: I'm fine to remove these

File: docs/guides/launch/docker.md
Comment (bcsherma) [reply docs/guides/launch/docker.md]: I think we should say that here (at the beginning) and also mention it in the tip

File: docs/guides/launch/launch-jobs.md
Comment (ngrayluna) [reply docs/guides/launch/launch-jobs.md]: Condensing this paragraph + making sentences shorter:

"Use W&B Launch to queue jobs for execution locally on your machine or on a compute resource.  Jobs pushed to a [queue](./create-queue.md) are picked up by an agent and executed using the compute and parameter configuration attached to the job. 

The following guide demonstrates how to add jobs to a launch queue."

Move this 
"Queues can be created by and for individual W&B users, or shared across W&B Teams. In either case, queue setup should be done once and then jobs can be added to the queue as needed.." to the "Create a queue" page.  

Let's change "user"  in  "Queues can be created by and for individual W&B users, or shared across W&B Teams. " to "member"? Or whatever is the correct definition regarding users, teams, members.

File: docs/guides/launch/getting-started.md
Comment (ngrayluna) [inline docs/guides/launch/getting-started.md]: To make this guide shorter/quicker.  We could move these thorough definitions to the "How it works" on the landing page. Here, we just have a quick, one-liner.

File: docs/guides/launch/intro.md
Comment (ngrayluna) [inline docs/guides/launch/intro.md:45]: Since we're here. Want to update "/images/launch/toggle_beta_flag.png"? 

https://wandb.atlassian.net/browse/DOCS-642

**Comment by bcsherma** [comment]
> This is looking good. Do we intend on having a page for each environment? We have "Docker" will we have "Kubernetes"? Sagemaker? We may want to think about consolidating this info into Create a Queue (for the runner) and "Start an Agent" for the environment to start an agent

@KyleGoyette @ngrayluna my plan with this layout is to have individual pages for general reference of how to work with the basic components of launch- jobs, agents, queues. In addition, we will have true developer guides that include reference documentation for specific setups. "Launch on Docker" contains info about how to set up docker queues and run agents that consume items from those queues. I will also add "Launch on Kubernetes" which will do the same for k8s. Launch on Sagemaker and Launch on Vertex will follow.

I think it's important that we have one page that someone can read to get started with a particular setup. And we can link to the generic queue/agent/job docs from those guides where appropriate.

File: docs/guides/launch/create-job.md
Comment (bcsherma) [reply docs/guides/launch/create-job.md]: Like change the title or the style of the callout? I like red flames one, but I think changing the text from danger to warning is good

File: docs/guides/launch/docker.md
Comment (bcsherma) [reply docs/guides/launch/docker.md:1]: I think the "launch on k8s" will have sections for deploying agents in EKS, GKE, and how kaniko works. Plus all the queue config stuff.

File: docs/guides/launch/create-job.md
Comment (ngrayluna) [reply docs/guides/launch/create-job.md]: Change it to non red flames. Product thinks it's too aggressive. :/

Comment (bcsherma) [reply docs/guides/launch/create-job.md]: Okay 😭 btw it also shows up on https://docs.wandb.ai/guides/reports/edit-a-report

**Comment by bcsherma** [comment]
I've addressed + resolved most of the comments from both of your reviews @ngrayluna @KyleGoyette (thank you for reviewing) and I also added the "launch on kubernetes" page with instructions on how to configure k8s queues and deploy agents in k8s. I will try to address all the rest of the comments but I also want to check on how mergeable this feels currently. I would really like to get this in this week.

**Comment by ngrayluna** [comment]
Ahoy  Ben! Thanks for putting all this work in. ❤️ 

What are folks' thoughts on grouping together the Launch on Docker and Launch on Kubernetes together? So the directory would look like:

| Getting started
.
.
.
| Start an agent
| Launch a run
|  new subchapter name
..... | Launch on Docker
......| Launch on Kubernetes
| Sweeps on Launch

**Comment by ngrayluna** [comment]
The last thing we need to do is add redirects to the data.json file. Any url that was changed or removed, needs to be added there.

**Comment by ngrayluna** [comment]
PS Hold off on merging until we have the redirects in + I give the green light. I have a W&B Server docs release that will be merged and go live tomorrow.

**Review by KyleGoyette — APPROVED** [review]

**Comment by bcsherma** [comment]
@ngrayluna thanks for all of the reviews! Will you handle the redirects? I'm not sure what the task there is exactly.

On the subject of the subchapter - I'm not sure what that means exactly, but would launch on docker and launch on k8s be separate pages still? Just organized together in the sidebar? I think that would be great but I don't want to combine the pages because we will keep adding more and the page will get cray

**Pull request merged**
